lib/matplotlib/axes/_axes.py

```diff
@@ -1083,10 +1083,10 @@ def hlines(self, y, xmin, xmax, colors=None, linestyles='solid',
         lines._internal_update(kwargs)
 
         if len(y) > 0:
-            minx = min(xmin.min(), xmax.min())
-            maxx = max(xmin.max(), xmax.max())
-            miny = y.min()
-            maxy = y.max()
+            minx = min(np.nanmin(xmin), np.nanmin(xmax))
```

Comment:
Since these are already combining/using masks above, would it make sense to add a `masked_invalid()` call earlier?

Reply:
I do not really know how masks actually works (which is a bit ironically, considering my comment on your PR about masks I did just before seeing this...), although I realize that it may be related. Just noted that this didn't work correctly when trying out the other fixes.

I may give it a do though, either here or as a follow-up.

Reply:
:+1: There is a `cbook.safe_masked_invalid` if that would help at all. Masking the invalid locations just ignores those in the computations (well the details are more involved, but that is the easy way of thinking about it).

Reply:
I've played around with it and cannot get it to work. The reason seems to be that the array has `dtype=object` (although it is float when created) and therefore I get runtime warnings on `min` and `max`.

Using `masked_invalid` gives a masked array with a fill_value of `?`, while `cbook.safe_masked_invalid` just returns an identical array.

If the array was `float` it would probably work though, but I have no idea where it is becoming `dtype=object`. Maybe related to `datetime` support?

(Not really clear to me why `nanmin` and `nanmax` is worse than adding masked conversions.)

Reply:
I don't think there is necessarily anything wrong with it if we are expecting nan's for some reason. To me, this just seemed like it was potentially hiding something else, which as you've noted here this only happens for `dtype=object`. If you are passing in a float array, I really don't think we should be making it an object array 😟  Perhaps this is because these error bars are generally used for categoricals so something is converting to objects fast because of that?

Reply:
We do this because we don't want to drop units.